### ISSUE TITLE
`getCameraForGeometry` very wrong values when map is tilted

### ISSUE THREAD

**Issue by westnordost** [maplibre/maplibre-native#2789]
**Describe the bug**
`getCameraForGeometry(Geometry geometry, int[] padding)` should return the necessary camera zoom and position to enclose the given geometry on the current camera rotation and tilt.

This works fine for any rotation, but with **tilt**, this sometimes calculates a zoom much much lower than would be appropriate. The decisive factor for how wrong the calculation turns out to be seems to be the **current camera position in relation to the position of the geometry**. See the video:

https://github.com/user-attachments/assets/ce12141f-af14-4824-8ad7-2ea8c5eabd60

**To Reproduce**
Reproducable with https://github.com/streetcomplete/StreetComplete/releases/tag/v59.0-alpha2

A minimum test with which this should be reproducible is to have a GeoJson layer with one geometry that focuses the camera on tap without changing rotation or tilt. Then, tilt the map and select it like in the video.

**Expected behavior**
Correct calculation for `getCameraForGeometry`

**Platform information:**
 - Android, any version
 - Version 11.2.0

**Additional context**
This is likely related to #2255, as surely the same algorithm is used to convert a lat-lon-pair to screen coordinates and back.

**Comment by Fabi755** [comment]
We have currently a similar issue.

1. Have a tilt (in all following steps)
2. Zoom in very near to the map
3. And then call `getCameraForGeometry`, the returned zoom is still the same as the current zoom.

---

I created a sample repo for this issue:
https://github.com/Fabi755/maplibre-getgeometry-issue

It needs around 5 times, until the right camera position is reached.

https://github.com/user-attachments/assets/7cbe5b45-c00c-4686-85ef-1e19969e3f9f

As I don't know anything about C++, would be nice if someone with knowledge can have a look to this issue.

**Comment by louwers** [comment]
Is this the same as #4039?

**Comment by Fabi755** [comment]
Yes, both functions `getCameraForGeometry` and `getCameraForLatLngBounds` using the [same native function `cameraForLatLngs`](https://github.com/maplibre/maplibre-native/blob/0592e4f0ad5475ce6f5e0c9bffcd2fac11cd9c56/src/mbgl/map/map.cpp#L254) (link seems broken: see line 254).
